packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java

```diff
@@ -113,7 +113,7 @@ public void createRuleInNonExistentFolder()
         restClient.authenticateUser(user).withCoreAPI().usingNode(nonExistentFolder).usingDefaultRuleSet().createSingleRule(ruleModel);
 
         restClient.assertStatusCodeIs(NOT_FOUND);
-        restClient.assertLastError().containsSummary("fake-id was not found");
+        restClient.assertLastError().containsSummary("The entity with id: fake-id which is a folder was not found");
```

Comment:
These messages are really misleading like this. The entity with id "fake-id" is definitely _not_ a folder.

Reply:
I've updated the messages after adding a new constructor to the Exception class. Also took the suggestion to also throw the original exception.